auth.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/openshift/origin/pkg/auth/authenticator/request/x509request"
```

Comment:
No. The reason for using this package is to enable more configurable auth handlers in future. Right now we only see a need for client cert auth but there may be call for other auth methods in future or more granular authzn policy. We've learnt the hard way that it's better to put the auth behind interfaces now in case we want to enhance the auth policy later.